edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/planners/HealthRestartPlanner.cs

```diff
@@ -355,7 +355,16 @@ DiffState ProcessDiff(ModuleSet desired, ModuleSet current)
 
             // Find the modules whose desired and runtime status are not the same
             IList<IRuntimeModule> updateStateChanged = currentRuntimeModules
-                .Where(m => m.DesiredStatus != m.RuntimeStatus).ToList();
+                .Where(m => m.DesiredStatus != m.RuntimeStatus && m.RuntimeStatus != ModuleStatus.Dead).ToList();
+
+            // Remove dead modules. The next deployment will recreate them if needed.
+            IList<IRuntimeModule> dead = currentRuntimeModules
+                .Where(m => m.RuntimeStatus == ModuleStatus.Dead).ToList();
+
+            foreach (IRuntimeModule deadModule in dead)
+            {
+                removed.Add(deadModule);
```

Comment:
I think the cleaner fix would be to return dead modules here, and in ProcessDesiredAndCurrentSets, remove them.

Reply:
Fixed